architecture/ambient/peer-authentication.md

```diff
@@ -0,0 +1,70 @@
+# PeerAuthentication Implementation in Ambient
+
+The PeerAuthentication resource defines policy for whether or not traffic to a given mesh workload must be mTLS encrypted (through Istio mTLS specifically). While the semantics for sidecars are [relatively well defined](https://istio.io/latest/docs/reference/config/security/peer_authentication/), the architectural differences of ambient workloads have implications for how PeerAuthentication is enforced in that context. This document describes those details.
+
+## PeerAuthentication and ztunnel
+
+The goal of ztunnel is to be a minimal L4 proxy, and as such, its xDS configuration is purposefully limited. In particular, ztunnel only supports 2 (custom) xDS resources: [`Workload`](../../pkg/workloadapi/workload.proto) and [`Authorization`](../../pkg/workloadapi/security/authorization.proto). As such, ztunnel does not receive `PeerAuthentication`s directly; when istiod detects a `PeerAuthentication` resource that targets an Ambient captured workload, it computes the effective policy for that workload (taking into account the mesh-wide -> namespace -> workload precedence rules) and sends that policy to ztunnel. The exact details of this conversion is out of scope for this document at the time of writing, but as an example, this `PeerAuthentication`:
```

Comment:
nit: technically its either 2 (address + Authorization) or 3 (workload + service + Authorization) depending on how you look at it...

Reply:
Yeah the proto naming is confusing me. All of the xDS code in pilot that I've seen refers to Workload, the protos and ztunnel and istiod say Workload, but in the ztunnel xDS client, we check for the Address type